conceptnet5/vectors/formats.py

```diff
@@ -1,85 +1,74 @@
+    """
     np.save(matrix_filename, table.values)
     save_index_as_labels(table.index, vocab_filename)
 
 
-def export_conceptnet_to_hyperwords(table, matrix_filename, vocab_filename, nrows):
-    vecs = []
-    labels = []
-    english_labels = [
-        standardized_uri('en', item)
-        for item in wordfreq.top_n_list('en', nrows * 2, 'large')
-    ]
-    count = 0
-    for label in english_labels:
-        if label in table.index:
-            labels.append(label.split('/')[-1])
-            vecs.append(get_vector(table, label))
-            count += 1
-            if count >= nrows:
-                break
-    np.save(matrix_filename, np.vstack(vecs))
-    save_index_as_labels(labels, vocab_filename)
-
-
-def export_plain_text(table, uri_file, file_base):
-    from ..vectors.query import VectorSpaceWrapper
-
-    def vec_to_text_line(label, vec):
-        cells = [label] + ['%4.4f' % val for val in vec]
-        return ' '.join(cells)
-
-    uri_main_file = gzip.open(file_base + '_uris_main.txt.gz', 'wt')
-    english_main_file = gzip.open(file_base + '_en_main.txt.gz', 'wt')
-    english_extra_file = gzip.open(file_base + '_en_extra.txt.gz', 'wt')
-    wrap = VectorSpaceWrapper(frame=table)
-
-    for line in open(uri_file, encoding='utf-8'):
-        uri = line.strip()
-        if uri.count('/') == 3 and get_uri_language(uri) in COMMON_LANGUAGES:
-            if uri in table.index:
-                vec = table.loc[uri].values
-                print(vec_to_text_line(uri, vec), file=uri_main_file)
-            else:
-                if not uri.startswith('/c/en') or '_' in uri:
-                    continue
-                vec = wrap.get_vector(uri)
-
-            if vec.dot(vec) == 0:
-                continue
+def vec_to_text_line(label, vec):
+    """
+    Output a labeled vector as a line in a fastText-style text format.
+    """
+    cells = [label] + ['%4.4f' % val for val in vec]
+    return ' '.join(cells)
 
-            if uri.startswith('/c/en/'):
-                label = uri[6:]
-                if uri in table.index:
-                    print(vec_to_text_line(label, vec), file=english_main_file)
-                else:
-                    print(vec_to_text_line(label, vec), file=english_extra_file)
 
-    uri_main_file.close()
-    english_main_file.close()
-    english_extra_file.close()
+def export_text(frame, filename, filter_language=None):
+    """
+    Save a semantic vector space as a fastText-style text file.
+
+    If `filter_language` is set, it will output only vectors in that language.
+    """
+    vectors = frame.values
+    index = frame.index
+    if filter_language is not None:
+        start_idx = index.get_loc('/c/%s/#' % filter_language, method='bfill')
+        try:
+            end_idx = index.get_loc('/c/%s0' % filter_language, method='bfill')
+        except KeyError:
+            end_idx = frame.shape[0]
+        frame = frame.iloc[start_idx:end_idx]
+        index = frame.index
```

Comment:
I think that would introduce a circular dependency between `conceptnet5.vectors.query` and `conceptnet5.vectors.formats`.